spec/components/search_spec.rb

```diff
@@ -8,6 +8,7 @@ def component_name
   it "renders a search box with default options" do
     render_component({})
     assert_select ".gem-c-search.gem-c-search--on-white"
+    assert_select "label[class='^=govuk-\!-margin-bottom-']", count: 0
```

Comment:
These "starts-with" attribute selectors aren't formatted quite right. They're passing because they're looking for nothing, but they're dud selectors so will always return nothing, thus making the tests redundant.

```suggestion
    assert_select "label[class^='govuk-\!-margin-bottom-']", count: 0
```

As a bit of extra evidence I did some local testing on this with a very rough test. This passed:

```
it "test" do
    render_component({})

    assert_select "div[class^='gem-c-']"
  end
```

This didn't:

```
it "test" do
    render_component({})

    assert_select "div[class='^gem-c-']"
  end
```

This applies also to lines 150 and 157.

Reply:
Good catch 🕵️‍♂️ 

This has been fixed for the search component in https://github.com/alphagov/govuk_publishing_components/pull/2462/commits/ba38459d46a9e48af46a603db64623174689dacc.

And there were other instances of this error which I've fixed in https://github.com/alphagov/govuk_publishing_components/pull/2463.